pkg/generic/generating_test.go

```diff
@@ -0,0 +1,116 @@
+package generic_test
+
+import (
+	"context"
+	"github.com/rancher/wrangler/v2/pkg/generic"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/rancher/wrangler/v2/pkg/apply"
+	fakeapply "github.com/rancher/wrangler/v2/pkg/apply/fake"
+	v1 "github.com/rancher/wrangler/v2/pkg/generated/controllers/core/v1"
+	fake2 "github.com/rancher/wrangler/v2/pkg/generic/fake"
+)
+
+func TestUniqueApplyForResourceVersion(t *testing.T) {
+	const n = 3
```

Comment:
nit: make this variable more descriptive
```suggestion
	const numOfHandlerCalls = 3
```